nsxt/data_source_nsxt_policy_vpc.go

```diff
@@ -0,0 +1,57 @@
+/* Copyright © 2024 Broadcom, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/bindings"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+)
+
+func dataSourceNsxtPolicyVPC() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceNsxtPolicyVPCRead,
+		Schema: map[string]*schema.Schema{
+			"id":                   getDataSourceIDSchema(),
+			"display_name":         getDataSourceDisplayNameSchema(),
+			"description":          getDataSourceDescriptionSchema(),
+			"path":                 getPathSchema(),
+			"context":              getContextSchema(true, false),
+			"site_info":            getSiteInfoSchema(),
+			"default_gateway_path": getPathSchema(),
+			"short_id": {
+				Type:     schema.TypeString,
+				Optional: true,
```

Comment:
According to the code this is fully computed, in which case `Optional: true` should be removed, so that terraform does not allow user to specify it

Reply:
This value is assigned by NSX if not set by user, and therefore has to be computed. It can also be set by the user (it should be unique - NSX will enforce that).
The UI doesn't allow updating it (on create there's the text "This field can be set at VPC creation time and cannot be edited later. Max 8 characters allowed.").
BTW there's an open issue for this in the Project.